src/Graphing/Enrich.hs

```diff
@@ -0,0 +1,230 @@
+  m (Graphing Dependency) -- graph with archive dependencies
+conanToArchives rootPath g =
+  -- If we do not have any conan dependencies, we do not
+  -- need to do any work!
+  case (null unableToTransformConanDep, transformedVendorDep) of
+    (True, Nothing) -> pure g
+    (False, _) -> fatal $ FailedToTransformConanDependency unableToTransformConanDep
+    (True, Just depsAndVendorDeps) -> do
+      let vendorDeps = NE.map snd depsAndVendorDeps
+
+      -- 1. We upload all vendor deps (transformed from conan deps),
+      -- since conan dependencies have to be scanned
+      -- before they can be referenced in graph as archive deps!
+      archiveLocators <-
+        licenseScanSourceUnit
+          -- always perform scan as vendor deps's
+          -- manifest are rarely updated with changes in practice
+          SkippingNotSupported
+          Nothing -- Ignore path filters
+          (FullFileUploads False)
+          rootPath
+          vendorDeps
+
+      -- 2. We make archive archive dependencies from locators, and original
+      -- conan dependencies.
+      let (failed, archiveDeps) =
+            partitionEithers $
+              map (locatorToArchiveDep allConanDeps) $
+                toList archiveLocators
+
+      unless (null failed) $
+        fatal $
+          FailedToTransformLocators failed
+
+      -- 3. We replace all conan dependencies with archive dependencies from
+      -- original graph. If we are unable to find twin of archive dep
+      -- (e.g. sourcing conan dep), we fail fatally!
+      case fromList <$> archiveToConanDep archiveDeps of
+        Left lonelyArchiveDeps -> fatal $ UnableToFindTwinOfArchiveDep lonelyArchiveDeps
+        Right registry -> pure $ gmap (\graphDep -> Map.findWithDefault graphDep graphDep registry) g
+  where
+    allDeps :: [Dependency]
+    allDeps = vertexList g
+
+    allConanDeps :: [Dependency]
+    allConanDeps = filter isConanDep allDeps
+
+    unableToTransformConanDep :: [Dependency]
+    unableToTransformConanDep = fst $ toVendorDeps allConanDeps
+
+    transformedVendorDep :: Maybe (NonEmpty (Dependency, VendoredDependency))
+    transformedVendorDep = nonEmpty . snd $ toVendorDeps allConanDeps
+
+    archiveToConanDep :: [Dependency] -> Either [Dependency] [(Dependency, Dependency)]
+    archiveToConanDep archiveDeps = sequence' $ map (findArchiveTwin archiveDeps) allConanDeps
+
+-- | True if dependency is a conan dependency, otherwise False.
+isConanDep :: Dependency -> Bool
+isConanDep d = dependencyType d == ConanType
+
+sequence' :: [Either a b] -> Either [a] [b]
+sequence' r = case partitionEithers r of
+  ([], xs) -> Right xs
+  (xs, _) -> Left xs
+
+toVendorDeps :: [Dependency] -> ([Dependency], [(Dependency, VendoredDependency)])
+toVendorDeps deps = partitionEithers $ map conanDepToVendorDep deps
+
+-- | Trasnforms Conan dependency to Vendor Dependency.
+conanDepToVendorDep :: Dependency -> Either Dependency (Dependency, VendoredDependency)
```

Comment:
I think the behaviour of `conanToVendoredDep`, `locatorToArchiveDep` and `findArchiveTwin` is just complicated enough that we should have either unit tests for them or tests for the unhappy path of `conanToArchives` that shows that they return a `Left` result when we expect them to